pyiron_atomistics/atomistics/job/atomistic.py

```diff
@@ -612,19 +612,30 @@ def get_final_structure(self):
     def _get_structure(self, frame=-1, wrap_atoms=True):
         if self.structure is None:
             raise AssertionError('Structure not set')
-        snapshot = self.structure.copy()
         if self.output.cells is not None:
             try:
-                snapshot.cell = self.output.cells[frame]
+                cell = self.output.cells[frame]
             except IndexError:
                 if wrap_atoms:
                     raise IndexError('cell at step ', frame, ' not found') from None
-                snapshot.cell = None
+                cell = None
         if self.output.indices is not None:
             try:
-                snapshot.indices = self.output.indices[frame]
+                indices = self.output.indices[frame]
             except IndexError:
-                pass
+                indices = None
+        species = self['output/structure/species']
+        if indices is not None and len(indices) != len(self.structure):
+            snapshot = Atoms(species=species, indices=indices,
+                             positions=np.zeros(indices.shape + (3,)), cell=cell, pbc=self.structure.pbc)
+        else:
+            snapshot = self.structure.copy()
+            if cell is not None:
+                snapshot.cell = cell
+            if indices is not None:
+                snapshot.indices = indices
+            if species is not None:
+                snapshot.species = species
```

Comment:
You can't set the species directly and have to use `set_species(species)`

Reply:
Sorry, I realize that this doesn't work too since the species list should be a list of `ChemicalElement` instances and not symbols. IMO since this line basically does nothing since we use the original species info, it's probably best to remove this line.